move submission -> running timing to k8s runner

danielnorberg opened the pull request:
Make this measurement less dependent on event transitions happening on the same scheduler instance in preparation for HA multiple instance scheduler setup.

NatashaL commented:
LGTM

fabriziodemaria commented:
+1

ulzha commented on styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java:
```diff
@@ -133,34 +136,36 @@
   private final int podDeletionDelaySeconds;
-  private final Clock clock;
+  private final Time time;
+  private final ConcurrentMap<String, Long> submissions = new ConcurrentHashMap<>();
```
Hard to read, have to guess the semantics until I read into usage. `submissionTimeByExecutionId` or a comment saying as much would be superb.

ulzha commented:
:+1:

danielnorberg commented:
Moved the actual timing code to `MetricsStats`. The `KubernetesDockerRunner` now just calls `reportSubmission()` and `reportRunning()`.

Also replaced the map holding the submission timestamps with a bounded guava cache to ensure that it doesn't grow without bound in case we don't observe pods running.
